src/lib/components/table/RenameTable.svelte

```diff
@@ -0,0 +1,68 @@
+<script lang="ts">
```

Comment:
@AdityaHegde @ericpgreen2 Does it make sense to have all Modal components inside `/modal` and have their names like `RenameTable` or `ExportError` without modal keyword?

Reply:
I think it should be in the `/table` (which we'll switch to `/source`) directory, where it can be colocated with other source operations (like CreateSource, DeleteSource, ShareSource).

Sure, I named it "RenameSourceModal" for now. I was originally thinking to keep it "implementation agnostic" because in the future I think it'd be nice to rename the source right in the text field in the assets pane, and not open a full modal to do it. But let's just go with "RenameSourceModal" for now – looks clean.